x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/sparse_vector_search.yml

```diff
@@ -0,0 +1,324 @@
+
+---
+"Test sparse_vector search with pruning config":
+  - do:
+      search:
+        index: index-with-sparse-vector
+        body:
+          query:
+            sparse_vector:
+              field: ml.tokens
+              inference_id: text_expansion_model
+              text: "octopus comforter smells"
+              prune: true
+              pruning_config:
+                tokens_freq_ratio_threshold: 4
+                tokens_weight_threshold: 0.4
+  - match: { hits.total.value: 4 }
+  - match: { hits.hits.0._source.source_text: "the octopus comforter smells" }
+
+---
+"Test sparse_vector search with pruning config that changes recall":
+  - do:
+      search:
+        index: index-with-sparse-vector
+        body:
+          query:
+            sparse_vector:
+              field: ml.tokens
+              inference_id: text_expansion_model
+              text: "octopus comforter smells"
+              prune: true
+              pruning_config:
+                tokens_freq_ratio_threshold: 2
+                tokens_weight_threshold: 0.9
+  - match: { hits.total.value: 1 }
+  - match: { hits.hits.0._source.source_text: "the octopus comforter smells" }
+
+---
+"Test named, boosted sparse_vector search with pruning config":
+  - do:
+      search:
+        index: index-with-sparse-vector
+        body:
+          query:
+            sparse_vector:
+              field: ml.tokens
+              inference_id: text_expansion_model
+              text: "octopus comforter smells"
+              pruning_config:
+                tokens_freq_ratio_threshold: 4
+                tokens_weight_threshold: 0.4
+  - match: { hits.total.value: 4 }
+  - match: { hits.hits.0._source.source_text: "the octopus comforter smells" }
+  - match: { hits.hits.0._score: 3.0 }
+
+  - do:
+      search:
+        index: index-with-sparse-vector
+        body:
+          query:
+            sparse_vector:
+              field: ml.tokens
+              inference_id: text_expansion_model
+              text: "octopus comforter smells"
+              prune: true
+              pruning_config:
+                tokens_freq_ratio_threshold: 4
+                tokens_weight_threshold: 0.4
+              _name: i-like-naming-my-queries
+              boost: 100.0
+  - match: { hits.total.value: 4 }
+  - match: { hits.hits.0._source.source_text: "the octopus comforter smells" }
+  - match: { hits.hits.0.matched_queries: [ "i-like-naming-my-queries" ] }
+  - match: { hits.hits.0._score: 300.0 }
+
+---
+"Test sparse_vector search with default pruning config":
+  - do:
+      search:
+        index: index-with-sparse-vector
+        body:
+          query:
+            sparse_vector:
+              field: ml.tokens
+              inference_id: text_expansion_model
+              text: "octopus comforter smells"
+              prune: true
+  - match: { hits.total.value: 4 }
+  - match: { hits.hits.0._source.source_text: "the octopus comforter smells" }
+
```

Comment:
This test result is no different than the one with no pruning at all, are we sure we are testing these paths? I would expect different hits or scores or something.